Admin/Remove-DuplicateEntriesFromIanaMappings.ps1

```diff
@@ -0,0 +1,205 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+#Requires -Version 3.0
+
+<#
+.SYNOPSIS
+    Removes duplicate entries from the IanaTimeZoneMappings.xml file which is used by Exchange Server
+.DESCRIPTION
+    Removes duplicate entries from the IanaTimeZoneMappings.xml file which is used by Exchange Server.
+    Duplicate entries can lead to exceptions and break functionalities on Microsoft Exchange Server such as processing .ics files.
+
+    For more information see: https://aka.ms/ExchangeIanaTimeZoneIssue
+.PARAMETER Server
+    The Exchange server that should be validated by the script. It also accepts values directly from the pipeline for seamless integration.
+.PARAMETER RestartServices
+    Specifies whether the following services should be restarted on the target system: W3SVC, WAS, MSExchangeTransport
+    Default value: $false
+.PARAMETER ScriptUpdateOnly
+    This optional parameter allows you to only update the script without performing any other actions.
+.PARAMETER SkipVersionCheck
+    This optional parameter allows you to skip the automatic version check and script update.
+.EXAMPLE
+    PS C:\> .\Remove-DuplicateEntriesFromIanaMappings.ps1 -Server exch1.contoso.com
+.EXAMPLE
+    PS C:\> Get-ExchangeServer | .\Remove-DuplicateEntriesFromIanaMappings.ps1
+.EXAMPLE
+    PS C:\> .\Remove-DuplicateEntriesFromIanaMappings.ps1 -Server exch1.contoso.com -RestartServices $true
+#>
+
+[CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess, ConfirmImpact = 'High')]
+param (
+    [Parameter(Mandatory = $false, ValueFromPipeline = $true, ParameterSetName = "Default")]
+    [string[]]$Server = $env:COMPUTERNAME,
+
+    [Parameter(Mandatory = $false, ParameterSetName = "Default")]
+    [bool]$RestartServices = $false,
+
+    [Parameter(Mandatory = $true, ParameterSetName = "ScriptUpdateOnly")]
+    [switch]$ScriptUpdateOnly,
+
+    [Parameter(Mandatory = $false, ParameterSetName = "Default")]
+    [switch]$SkipVersionCheck
+)
+
+begin {
+    . $PSScriptRoot\..\Shared\Confirm-Administrator.ps1
+    . $PSScriptRoot\..\Shared\Get-ExSetupFileVersionInfo.ps1
+    . $PSScriptRoot\..\Shared\Invoke-ScriptBlockHandler.ps1
+    . $PSScriptRoot\..\Shared\ScriptUpdateFunctions\GenericScriptUpdate.ps1
+
+    Write-Verbose "PowerShell version: $($PSVersionTable.PSVersion)"
+
+    $scriptBlock = {
+        param(
+            $PerformServiceRestart,
+            $LocalComputerName
+        )
+
+        . $PSScriptRoot\..\Diagnostics\HealthChecker\Analyzer\Test-IanaTimeZoneMapping.ps1
```

Comment:
Move to `Shared` as discussed in previous PR.

Reply:
Addressed with next iteration.